docs/src/components/hardware-list.tsx

```diff
@@ -0,0 +1,156 @@
+import React from "react";
+
+import {
+  Board,
+  HardwareMetadata,
+  Interconnect,
+  Shield,
+} from "../hardware-metadata";
+
+interface HardwareListProps {
+  items: HardwareMetadata[];
+}
+
+function itemIds(item: HardwareMetadata) {
+  if (item.type == "board" || item.type == "shield") {
+    let nodes = (item.siblings || [item.id])
+      .map<JSX.Element | string>((id) => <code key={id}>{id}</code>)
+      .reduce(
+        (prev, curr, index) => [...prev, index > 0 ? " and " : "", curr],
+        [] as Array<string | JSX.Element>
+      );
+    return <span key={item.id}>{nodes}</span>;
+  } else {
+    return <code key={item.id}>{item.id}</code>;
+  }
+}
+
+function typeLabel(type: HardwareMetadata["type"]): string {
+  switch (type) {
+    case "board":
+      return "Board: ";
+    case "shield":
+      return "Shield: ";
+    case "interconnect":
+      return "Interconnect: ";
+  }
+}
+
+function HardwareLineItem({ item }: { item: HardwareMetadata }) {
+  return (
+    <li>
+      <a href={item.url}>{item.name}</a> ({typeLabel(item.type)} {itemIds(item)}
+      )
+    </li>
+  );
+}
+
+function mapInterconnect({
+  interconnect,
+  boards,
+  shields,
+}: {
+  interconnect?: Interconnect;
+  boards: Array<Board>;
+  shields: Array<Shield>;
+}) {
+  if (!interconnect) {
+    return null;
+  }
+
+  return (
+    <div key={interconnect.id}>
+      <h4>{interconnect.name} Keyboards</h4>
+      {interconnect.description && <p>{interconnect.description}</p>}
+      <h5>Boards</h5>
+      <ul>
+        {boards.map((s) => (
+          <HardwareLineItem key={s.id} item={s} />
+        ))}
+      </ul>
+      <h5>Shields</h5>
+      <ul>
+        {shields.map((s) => (
+          <HardwareLineItem key={s.id} item={s} />
+        ))}
+      </ul>
+    </div>
+  );
+}
+
+function HardwareList({ items }: HardwareListProps) {
+  let grouped: {
+    onboard: Array<Board>;
+    interconnects: {
+      [key: string]: {
```

Comment:
Repeated type from lines 53-55. Should probably be broken out into an interface. Then you can also simplify this to

```
interconnects: Record<string, Interconnect>;
```

Reply:
Indeed. Thanks!